lib/Doctrine/Common/Annotations/PsrCachedReader.php

```diff
@@ -0,0 +1,237 @@
+<?php
+
+namespace Doctrine\Common\Annotations;
+
+use Psr\Cache\CacheItemPoolInterface;
+use ReflectionClass;
+use ReflectionMethod;
+use ReflectionProperty;
+use Reflector;
+
+use function array_map;
+use function array_merge;
+use function assert;
+use function filemtime;
+use function max;
+use function rawurlencode;
+use function time;
+
+/**
+ * A cache aware annotation reader.
+ */
+final class PsrCachedReader implements Reader
+{
+    /** @var Reader */
+    private $delegate;
+
+    /** @var CacheItemPoolInterface */
+    private $cache;
+
+    /** @var bool */
+    private $debug;
+
+    /** @var array<string, array<object>> */
+    private $loadedAnnotations = [];
+
+    /** @var int[] */
+    private $loadedFilemtimes = [];
+
+    public function __construct(Reader $reader, CacheItemPoolInterface $cache, bool $debug = false)
```

Comment:
Given this line:
https://github.com/doctrine/orm/blob/4e0f6837d04306417640c6be1ab5a20f9cb9e5d6/lib/Doctrine/ORM/Configuration.php#L165

I think we should make $cache nullable, and make the implem deal with having not PSR cache. That would free doctrine/orm from requiring any psr/cache implementation when this line will go away.

Reply:
And to make `ensureProductionSettings()` useful, we should add a method that returns the PsrCache instance, to be used in:
https://github.com/doctrine/orm/blob/4e0f6837d04306417640c6be1ab5a20f9cb9e5d6/lib/Doctrine/ORM/Configuration.php#L374

Reply:
(forget about my last comment)

Reply:
how would a PsrCachedReader work without a PSR-6 implementation (and so no CachePool and no CacheItem) ?

Reply:
there is an internal array cache, so it would do something useful :)